nvme.c

```diff
@@ -925,22 +929,24 @@ int __id_ctrl(int argc, char **argv, struct command *cmd, struct plugin *plugin,
 	if (cfg.human_readable)
 		flags |= HUMAN;
 
-	err = nvme_identify_ctrl(fd, &ctrl);
-	if (!err) {
+	err = nvme_ctrl_get_by_fd(fd, &ctrl, &err_msg);
```

Comment:
The caller of `nvme_ctrl_get_by_fd()` should do the free. Let me add some simple doc for them.